python_modules/dagster/dagster/core/execution/plan/plan.py

```diff
@@ -368,6 +389,28 @@ def _build_from_sorted_solids(
                 self.set_output_handle(output_handle, step_output_handle)
 
 
+def get_root_graph_input_source(
+    plan_builder: _PlanBuilder,
+    input_name: str,
+) -> FromRootInputConfig:
+
+    input_config = plan_builder.resolved_run_config.inputs
+
+    if input_config and input_name in input_config:
+        return FromRootInputConfig(input_name=input_name)
+
+    # Otherwise we throw an error.
+    raise DagsterInvariantViolationError(
```

Comment:
I think this needs a special case for `Nothing` like `get_step_input_source` does